### PR TITLE
test_connection: use lighter-weight mock to test s3:// channels

### PR DESCRIPTION
**Pull request by dholth**
### Description

Since conda's usage of boto3 is limited, use a simpler mock server.

I experimented with removing `{"CONDA_USE_ONLY_TAR_BZ2": "True",` to use `.conda`, but we get a seek error on some CI runners. That can be caused by tarfiles with duplicate entries like those we used to include in the conda-build or conda-index test suite, but I don't see which those might be here.

The test makes these requests only. They are normal HTTP GET and HEAD requests with s3 auth headers added. It does not look for other "s3 API" features on the package server.
```
% CONDA_DEBUG=1 pytest -k s3_server_with -vv -s 2>&1 | grep :make_request
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=GetObject) with params: {'url_path': '/test/linux-64/repodata.json', 'query_string': {}, 'method': 'GET', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/linux-64/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x107b67e50>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=HeadObject) with params: {'url_path': '/test/linux-64/repodata.json', 'query_string': {}, 'method': 'HEAD', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/linux-64/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x107b67e50>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=GetObject) with params: {'url_path': '/test/linux-64/repodata.json', 'query_string': {}, 'method': 'GET', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/linux-64/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x107b67e50>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=GetObject) with params: {'url_path': '/test/noarch/repodata.json', 'query_string': {}, 'method': 'GET', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/noarch/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x110372350>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=HeadObject) with params: {'url_path': '/test/noarch/repodata.json', 'query_string': {}, 'method': 'HEAD', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/noarch/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x110372350>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
DEBUG botocore.endpoint:make_request(114): Making request for OperationModel(name=GetObject) with params: {'url_path': '/test/noarch/repodata.json', 'query_string': {}, 'method': 'GET', 'headers': {'User-Agent': 'Boto3/1.24.28 Python/3.11.5 Darwin/22.6.0 Botocore/1.27.59 Resource'}, 'body': b'', 'url': 'https://127.0.0.1:52619/test/noarch/repodata.json', 'context': {'client_region': 'us-east-1', 'client_config': <botocore.config.Config object at 0x110372350>, 'has_streaming_input': False, 'auth_type': None, 'signing': {'bucket': 'test'}}}
```

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: tests/data/conda_format_repo/linux-64/current_repodata.json [hunk 1]
```diff
@@ -27,10 +13,10 @@
       "legacy_bz2_md5": "2e5b81671cc2b53177b94b258b1658b2",
       "license": "zlib",
       "license_family": "Other",
-      "md5": "2e5b81671cc2b53177b94b258b1658b2",
+      "md5": "cb23839f06da92ec92c6d2d2d5e0cb4a",
       "name": "zlib",
-      "sha256": "1f07873d6aa0cd18760a7b215d9956f7fa63c2f2761b03f293c659fe982923fa",
-      "size": 122456,
+      "sha256": "f33afff2144e1d6c4cf2e3e184937db8bf825b4537df74662aea5ee207b6a31b",
```
Comment (dholth) [inline tests/data/conda_format_repo/linux-64/current_repodata.json:18]: These hashes didn't match the test repository contents.

We'll see if any tests require the missing `libgcc-ng`.

File: tests/gateways/test_connection.py [hunk 2]
```diff
@@ -106,7 +107,60 @@ def test_s3_server(minio_s3_server):
...
+        False,  # verify
+        endpoint_url,  # endpoint_url
+        "minioadmin",  # aws_access_key_id
+        "minioadmin",  # aws_secret_access_key
+        None,  # aws_session_token
+        Config(signature_version="s3v4"),  # config
+    )
+    with pytest.raises(CondaExitZero):
+        with patch.object(
+            boto3.session.Session.resource, "__defaults__", patched_defaults
```
Comment (dholth) [inline tests/gateways/test_connection.py:155]: This patch is the same as in `test_s3_server` using minio except for the url. It is used to point boto3 at a non-AWS server.

File: tests/data/conda_format_repo/osx-64/repodata.json [hunk 3]
```diff
@@ -25,10 +25,10 @@
       "depends": [],
       "license": "zlib",
       "license_family": "Other",
-      "md5": "5eab2de4933e1914f86d1586a8a40c39",
+      "md5": "a1b03ba4ff40c83e4eb9c8fae40315a7",
       "name": "zlib",
-      "sha256": "9b73eb4e3197eccb485250709c6f4782a83a8688d5b8251a212251c451d9b9ad",
-      "size": 106948,
+      "sha256": "857293f389b188ba1fb02401a873fca5098db612ce6646c684efee77ad807f77",
```
Comment (dholth) [inline tests/data/conda_format_repo/osx-64/repodata.json:30]: I notice that previously the sha256 sum incorrectly matched that of the `.tar.bz2` version.

**Review by kenodegard — CHANGES_REQUESTED** [review]
this looks reasonable

we should combine `test_s2_server` and `test_s2_server_with_mock` before merging

**Comment by dholth** [comment]
> this looks reasonable
> 
> we should combine `test_s2_server` and `test_s2_server_with_mock` before merging

I thought it might be a lot to combine both the new test, and all the work removing the old test and its associated scripts and caches in the same PR.

**Comment by jezdez** [comment]
I've decided to go with #13084 for now to simplify our CI config, and then review this one again to make sure we're not risking test coverage.

File: tests/cli/test_main_search.py [hunk 4]
```diff
@@ -143,7 +143,9 @@ def test_search_envs_info(conda_cli: CondaCLIFixture):
 
 @pytest.mark.integration
 def test_search_envs_json(conda_cli: CondaCLIFixture):
-    stdout, _, _ = conda_cli("search", "--envs", "--json", "conda")
+    search_for = "python"
+    stdout, _, _ = conda_cli("search", "--envs", "--json", search_for)
```
Comment (dholth) [inline tests/cli/test_main_search.py:147]: python was more likely to be found than conda

**Comment by beeankha** [comment]
pre-commit.ci autofix

**Review by beeankha — APPROVED** [review]

**Pull request merged**